exchange/bitswap/wantmanager.go

```diff
@@ -168,11 +168,17 @@ func (mq *msgQueue) runQueue(ctx context.Context) {
 }
 
 func (pm *WantManager) Connected(p peer.ID) {
-	pm.connect <- p
+	select {
+	case pm.connect <- p:
+	case <-pm.ctx.Done():
+	}
 }
```

Comment:
Ah, looks like they're not used at all now (except in tests), but may
be used in the future by the p2p implementation [1].  Although I can't
square that with hanging goroutines in a manually-killed daemon.
Maybe a stuck test?

[1]: https://github.com/ipfs/go-ipfs/blob/41785f3b590156e05d01500d0c072b516f000a5a/p2p/net/interface.go#L152-L154